From 81e2122d156ad23028610f561ee6a4ef5eef908b Mon Sep 17 00:00:00 2001
From: William Douglas <william.douglas@intel.com>
Date: Tue, 23 May 2023 11:10:43 -0700
Subject: [PATCH] Add makefile targets for Clear Linux OS

install target is directly adapted from the mock.spec in the upstream
source tree.

Ported-by: William Douglas <william.douglas@intel.com>
Signed-off-by: Patrick McCarty <patrick.mccarty@intel.com>
Signed-off-by: William Douglas <william.douglas@intel.com>
---
 Makefile | 36 ++++++++++++++++++++++++++++++++++++
 1 file changed, 36 insertions(+)

diff --git a/Makefile b/Makefile
index 209074c..f290bf1 100644
--- a/Makefile
+++ b/Makefile
@@ -1,4 +1,40 @@
 # vim:noexpandtab:autoindent:tabstop=8:shiftwidth=8:filetype=make:nocindent:tw=0:
+SHELL = /bin/bash
+VERSION ?= $(shell rpmspec --srpm -q --qf '%{VERSION}' mock/mock.spec)
+
+all:
+	@echo nothing to do
+
+install:
+	cd mock; \
+	pysite=$$(python3 -c 'import site; print(site.getsitepackages()[0])'); \
+	for i in py/mockbuild/constants.py; do \
+		sed -i 's|^\(VERSION\).*|\1 = "'$(VERSION)'"|' $$i; \
+		sed -i 's|^\(SYSCONFDIR\).*|\1 = "'/etc'"|' $$i; \
+		sed -i 's|^\(PYTHONDIR\).*|\1 = "'$$pysite'"|' $$i; \
+		sed -i 's|^\(PKGPYTHONDIR\).*|\1 = "'$$pysite/mockbuild'"|' $$i; \
+	done; \
+	for i in docs/mock.1 docs/mock-parse-buildlog.1; do \
+		sed -i 's|@VERSION@|'$(VERSION)'|' $$i; \
+	done; \
+	install -d $(DESTDIR)/usr/bin; \
+	install -d $(DESTDIR)/usr/libexec/mock; \
+	install mockchain $(DESTDIR)/usr/bin/mockchain; \
+	install py/mock-parse-buildlog.py $(DESTDIR)/usr/bin/mock-parse-buildlog; \
+	install py/mock.py $(DESTDIR)/usr/bin/mock; \
+	install create_default_route_in_container.sh $(DESTDIR)/usr/libexec/mock/; \
+	install -d $(DESTDIR)/usr/share/pam.d; \
+	cp -a etc/pam/* $(DESTDIR)/usr/share/pam.d/; \
+	install -d $(DESTDIR)/usr/share/defaults/mock; \
+	cp -a etc/mock/* $(DESTDIR)/usr/share/defaults/mock/; \
+	cp -a docs/site-defaults.cfg $(DESTDIR)/usr/share/defaults/mock/; \
+	install -d $(DESTDIR)/usr/share/bash-completion/completions/; \
+	cp -a etc/bash_completion.d/* $(DESTDIR)/usr/share/bash-completion/completions/; \
+	install -d $(DESTDIR)/$$pysite; \
+	cp -a py/mockbuild $(DESTDIR)/$$pysite/; \
+	install -d $(DESTDIR)/usr/share/man/man1; \
+	cp -a docs/mock.1 docs/mock-parse-buildlog.1 $(DESTDIR)/usr/share/man/man1/;
+
 check:
 	./integration-tests/runtests.sh
 
-- 
2.40.1

